swap_meet/vendor.py

```diff
@@ -1,2 +1,45 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+            
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, friend, my_item, their_item):  
+        if my_item not in self.inventory or their_item not in friend.inventory:
```

Comment:
... I realize now I was reading these comments backwards. Index makes more sense since that check is constant, whereas remove is O(n) since we need to shift something as you said later.